db/model/generatortemplate.js

```diff
@@ -0,0 +1,187 @@
+      type: 'string',
+    },
+    toUrl: {
+      description: 'Optional toUrl given by the author',
+      type: 'string',
+    },
+    headers: {
+      description: 'Optional connection header information',
+      type: 'object',
+    },
+  },
+};
+
+const ctxDefRequiredProps = ['description'];
+
+module.exports = function user(seq, dataTypes) {
+  const GeneratorTemplate = seq.define('GeneratorTemplate', {
+    description: {
+      type: dataTypes.TEXT,
+    },
+    helpEmail: {
+      type: dataTypes.STRING(constants.fieldlen.email),
+      validate: { isEmail: true },
+    },
+    helpUrl: {
+      type: dataTypes.STRING(constants.fieldlen.url),
+      validate: { isUrl: true },
+    },
+    id: {
+      type: dataTypes.UUID,
+      primaryKey: true,
+      defaultValue: dataTypes.UUIDV4,
+    },
+    name: {
+      type: dataTypes.STRING(constants.fieldlen.normalName),
+      allowNull: false,
+      validate: {
+        is: constants.nameRegex,
+      },
+    },
+    version: {
+      defaultValue: '1.0.0',
+      type: dataTypes.STRING(constants.fieldlen.shortish),
+      validate: {
+        is: constants.versionRegex,
+      },
+    },
+    bulk: {
+      type: dataTypes.BOOLEAN,
+      defaultValue: false,
+    },
+    keywords: {
+      type: dataTypes.ARRAY(dataTypes.STRING(constants.fieldlen.normalName)),
+      allowNull: true,
+      defaultValue: constants.defaultArrayValue,
+    },
+    author: {
+      type: dataTypes.JSON,
+      allowedNull: false,
+      validate: {
+        validateObject(value) {
+          common.validateObject(value, authorSchema);
+        },
+      },
+    },
+    repository: {
+      type: dataTypes.JSON,
+      allowNull: true,
+      validate: {
+        validateObject(value) {
+          common.validateObject(value, repositorySchema);
+        },
+      },
+    },
+    connection: {
+      type: dataTypes.JSON,
+      allowNull: true,
+      validate: {
+        validateObject(value) {
+          common.validateObject(value, connectionSchema);
+        },
+      },
+    },
+    contextDefinition: {
+      type: dataTypes.JSON,
+      allowNull: true,
+      validate: {
+        validateObject(value) {
+          common.validateContextDef(value, ctxDefRequiredProps);
+        },
+      },
+    },
+    transform: {
+      type: dataTypes.TEXT,
+      allowNull: false,
+    },
+    isPublished: {
+      type: dataTypes.BOOLEAN,
+      allowNull: false,
+      defaultValue: false,
+    },
+  }, {
+    classMethods: {
+      getGeneratortemplateAssociations() {
+        return assoc;
+      },
+
+      postImport(models) {
+        assoc.createdBy = GeneratorTemplate.belongsTo(models.User, {
+          foreignKey: 'createdBy',
+        });
+      },
+    },
+
+    indexes: [
+      {
+        name: 'GTNameVersion',
+        unique: true,
+        fields: [
+          'name',
+          'version',
+        ],
+      },
+    ],
+    paranoid: true,
```

Comment:
I don't think we need paranoid--we're not going to allow deletes of generator templates because we have to preserve them for audit history.
Also, we should add a beforeDelete hook and throw an error if you try to delete one.